Makefile

```diff
@@ -392,20 +392,29 @@ devinstall-systemd: $(wildcard systemd/*)
 	@systemctl daemon-reload
 
 .PHONY: devinstall-hax
-devinstall-hax: HAX_INSTALL_CMD = $(SETUP_PY) develop --prefix $(DESTDIR)/$(PREFIX)
+# Don't specify --prefix $(DESTDIR)/$(PREFIX) since we do want to use our
+# virtualenv folder instead. Generated executables will point to python binary
+# from our virtualenv via shebang. Those executables will be put to /opt/seagate
+# as symlinks. So it is critical to have virtualenv folder populated.
+devinstall-hax: HAX_INSTALL_CMD = $(SETUP_PY) develop
 devinstall-hax: export PYTHONPATH = $(DESTDIR)/$(PREFIX)/lib/python3.$(PY3_VERSION_MINOR)/site-packages
 devinstall-hax: hax/requirements.txt $(HAX_EGG_LINK)
 	@$(call _info,Installing hax development dependencies)
-	@$(PIP) install --ignore-installed --prefix $(DESTDIR)/$(PREFIX) \
+	@$(PIP) install --ignore-installed \
 					--requirement <(sed -ne '/^#:runtime-requirements:/,$$p' $<)
+	@$(call _info,Creating symlinks for hax executables)
+	@ln -v -sf $(PY_VENV_DIR)/bin/hax $(DESTDIR)/$(PREFIX)/bin
+	@ln -v -sf $(PY_VENV_DIR)/bin/q $(DESTDIR)/$(PREFIX)/bin
 
 .PHONY: devinstall-miniprov
-devinstall-miniprov: MP_INSTALL_CMD = $(SETUP_PY) develop --prefix $(DESTDIR)/$(PREFIX)
+devinstall-miniprov: MP_INSTALL_CMD = $(SETUP_PY) develop
 devinstall-miniprov: export PYTHONPATH = $(DESTDIR)/$(PREFIX)/lib/python3.$(PY3_VERSION_MINOR)/site-packages
-devinstall-miniprov: provisioning/miniprov/requirements.txt $(HAX_EGG_LINK)
+devinstall-miniprov: provisioning/miniprov/requirements.txt $(MP_EGG_LINK)
 	@$(call _info,Installing miniprov development dependencies)
-	@$(PIP) install --ignore-installed --prefix $(DESTDIR)/$(PREFIX) \
+	@$(PIP) install --ignore-installed \
 					--requirement <(sed -ne '/^#:runtime-requirements:/,$$p' $<)
+	@$(call _info,Creating symlinks for mini-provisioner executables)
+	@ln -v -sf $(PY_VENV_DIR)/bin/hare_setup $(DESTDIR)/$(PREFIX)/bin
```

Comment:
#1529 creates link for `install-miniprov` target as well, do we need to incorporate that here? or we can merge #1529.

Reply:
I'm not sure that the fix in #1529 is correct. Please see my comment there.